### DISCUSSION TITLE
Issues with nordvpn wireguard and transmission

### DISCUSSION THREAD

**Discussion by mohkale** [transmission/transmission#2363] [General]
This is less an issue, more a discussion. Over the weekend I tried switching my nordvpn backend from openvpn to wireguard and I noticed when using transmission that I got lazy checksum validation errors for almost every other downloaded piece, slowing downloading to a crawl. Switching back to openvpn has fixed the issue so far, but I'm interested in what ither peoples experiences with wireguard have been.

**Comment by ckerr** [comment]
I believe most BitTorrent clients have decreased performance when using a VPN -- that's just going to be intrinsic to the nature of what you're doing. All of those peer round trips now have a new hop that may be halfway around the world.

If there is anything particular to Transmission's performance that could be improved, I would be happy to review PRs.

**Comment by mohkale** [comment]
@ckerr I'm aware of the degrade in performance with a VPN, but the issue I described above is probably closer to a bug in tranmission or wireguard (more likely wireguard). Switching back to an OpenVPN implementation works perfectly (albeit with a slight performance hit). To put it another way when using wireguard the pieces for a download failed repeatedly and after transmission reached around 90% done with a torrent, if I verified it again the completion-rate dropped back to 15% (I surmise because 75% of the downloaded pieces also failed but weren't checksumed lazily). I plan to investigate more at some point, but wanted to have an open discussion in case anyone has had similair experiences with other VPN providers or wireguard. :smile:.

**Comment by bjlockie** [comment]
> This is less an issue, more a discussion. Over the weekend I tried switching my nordvpn backend from openvpn to wireguard and I noticed when using transmission that I got lazy checksum validation errors for almost every other downloaded piece

Try 'sudo apt install wireguard'.
I don't know what's in that package but it helped me.

**Comment by mohkale** [comment]
I'm on arch and having wireguard installed wasn't exactly the issue here (internet worked fine through wireguard), but thanks for the advice all the same.

**Comment by minexew** [comment]
Hey, have you ever managed to resolve this?
I get a lot of
```
transmission-daemon: magnet status: checksum passed 0, metainfo parsed 0 (torrent-magnet.c:358)
```
in the logs, wondering if it is the same problem.

**Reply by mohkale** [reply]
For completed torrents? No, I'm not sure why but I switched back to openvpn and eventually to wireguard and then the issue was no longer their.
